setup.go

```diff
@@ -0,0 +1,831 @@
+		`\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]` +
+		`|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:` +
+		`(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|` +
+		`\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])`
+
+	// Default constants
+	defaultServerIP      = "127.0.0.1"
+	defaultServerURL     = "https://docker.mender.io"
+	defaultInventoryPoll = 28800 // NOTE: If changing these integer default
+	defaultRetryPoll     = 300   //       values, make sure to update the
+	defaultUpdatePoll    = 1800  //       corresponding prompt texts below.
+	demoInventoryPoll    = 5
+	demoRetryPoll        = 30
+	demoUpdatePoll       = 5
+	hostedMenderURL      = "https://hosted.mender.io"
+
+	// Prompt constants
+	promptWizard = "Mender Client Setup\n" +
+		"===================\n\n" +
+		"Setting up the Mender client daemon: the daemon will " +
+		"regularly poll the server to check for updates and report " +
+		"its inventory data.\nGet started by first configuring the " +
+		"device type and settings for communicating with the server:"
+	promptDone = "Mender is setup successfully. You may now start the " +
+		"daemon by running the `daemon` command."
+	promptDoneRunDaemon = "Starting Mender client…\n" + // NOTE: format
+		"Your device should show up at %s within the next few " +
+		"minutes.\nLog in to your Mender UI at %s to accept it.\n"
+	promptDeviceType   = "Device type (for example raspberrypi3-raspbian): "
+	promptConfirmDev   = "Confirm devicetype as %s? [Y/n] " // NOTE: format
+	promptHostedMender = "\nAre you connecting this device to Hosted Mender? [Y/n] "
+	promptCredentials  = "\nEnter your Hosted Mender credentials"
+	promptDemoMode     = "\nDemo mode uses short poll intervals and assumes the " +
+		"default demo server setup. (Recommended for testing.)\n" +
+		"Do you want to run the client in demo mode? [Y/n] "
+	promptServerIP = "\nSet the IP of the Mender Server: [" +
+		defaultServerIP + "] "
+	promptServerURL = "\nSet the URL of the Mender Server: [" +
+		defaultServerURL + "] "
+	promptServerCert = "\nSet the location of the certificate of the " +
+		"server; leave blank if using http (not recommended) or a " +
+		"certificate from a known authority " +
+		"(filepath, for example /etc/mender/server.crt): "
+	promptUpdatePoll = "\nSet the update poll interval - the frequency with " +
+		"which the client will send an update check request to the " +
+		"server, in seconds: [1800]" // (defaultUpdatePoll)
+	promptRetryPoll = "\nSet the retry poll interval - the frequency with " +
+		"which the client tries to communicate with the server (note: " +
+		"the client may attempt more often initially based on the " +
+		"previous intervals, but will fall back to this value if the" +
+		"server is busy) [300]" // (defaultRetryPoll)
+	promptInventoryPoll = "Set the inventory poll interval - the " +
+		"frequency with which the client will send inventory data to " +
+		"the server, in seconds: [28800]" // (defaultInventoryPoll)
```

Comment:
I tried utilizing the default values above, but it turns out that if you cast the integer constants to strings it becomes a corrupted unicode character probably because the compiler interprets the value as an int without resolving the string context.
Alternatively I could make the texts variables and use strconv.Atoi, but I'd rather keep them as constants to avoid unnecessary imports.

Reply:
Yep. That is more or less what I thought. Hence I am ok with the hardcoded :)